### DISCUSSION TITLE
Programmatically generate page links?

### DISCUSSION THREAD

**Discussion by farleykr** [markdoc/markdoc#21] [Q&A] [answered]
I noticed that in the SideNav component on the docs site, the links are generated from an array that contains all the information.  Is there a preferred way to programmatically generate links?

**Comment by mfix-stripe** [comment, marked as answer]
@farleykr I think it depends on your implementation. You could scan through all the pages at build time to create a "content manifest" (as we do [here](https://github.com/markdoc/docs/blob/main/examples/react-nodejs/createContentManifest.js)), and use that data to generate the side nav. 

However, typically you want more control over the order of the side nav items, as well as which pages show up, so hard coding it somewhere is probably your best bet (we do something similar at Stripe as well).

**Reply by farleykr** [reply]
Thanks!  It's helpful to know that's how you do it at Stripe too.

**Reply by anythingwithawire** [reply]
So would this be the path generate an index, or the elusive concordance?

**Reply by StarpTech** [reply]
Hi, we are run into the same question. We use the next.js integration. The downside of this approach is that we need to run it beside the next.js build process when modifying content. Is there built in support planned for the next.js plug in?

In your docs, you are maintaining the list manually https://github.com/markdoc/docs/blob/main/components/Shell/SideNav.js#L5 which doesn't scale.

**Reply by nkohari** [reply]
For what it's worth, we also manually curate the sidebar navigation at Stripe! In practice, the additional effort is worth it, because the navigation has such a large impact on your information architecture. It's fairly easy to generate a list of all of the documents in your corpus (the content manifest that mfix was describing) but deciding on the right ordering usually requires manual effort.

**Reply by StarpTech** [reply]
Hi @nkohari thanks for the context. I think this depends on the use case. I use markdoc for a blog. I need the list of pages to build links and indexes. It's also more about the integrated DX when using markdoc with Next.js. Right now, you need to watch for changes in a separate process and load the result synchronously in your `_app`. This is a no-brainer but the integration could be much better.

**Comment by timfee** [comment]
This use case reminds me of Contentlayer; they're currently [discussing support for Markdoc](https://github.com/contentlayerdev/contentlayer/issues/228) over there, and I think this could be an epic collab :)

**Reply by mfix-stripe** [reply]
@timfee I agree! Please keep us looped in if ContentLayer needs anything. Or, open a new ["Ideas" Discussion](https://github.com/markdoc/markdoc/discussions/categories/ideas)
